dali/pipeline/operators/reader/loader/file_loader.h

```diff
@@ -38,12 +39,21 @@ void assemble_file_list(const std::string& path, int label,
   DIR *dir = opendir(path.c_str());
   struct dirent *entry;
 
+  const std::vector<std::string> valid_extensions({".jpg", ".jpeg", ".png", ".bmp"});
+
   while ((entry = readdir(dir))) {
     std::string full_path = path + "/" + std::string{entry->d_name};
     struct stat s;
     stat(full_path.c_str(), &s);
     if (S_ISREG(s.st_mode)) {
-      file_label_pairs->push_back(std::make_pair(full_path, label));
+      std::transform(full_path.begin(), full_path.end(), full_path.begin(), ::tolower);
+      for (const std::string& s : valid_extensions) {
+        size_t pos = full_path.rfind(s);
+        if (pos != std::string::npos && pos + s.size() == full_path.size()) {
+          file_label_pairs->push_back(std::make_pair(full_path, label));
```

Comment:
Now lower case path is added to list, but FS is case sensitive and if there is any uppercase in the real path it would fail opening it.

Reply:
Facepalm. Yup, I will change that on Monday.

Reply:
Happens:)